### DISCUSSION TITLE
DataGrid - Validator using another property of the EditItem

### DISCUSSION THREAD

**Discussion by paulosdfigueiredo** [Megabit/Blazorise#1638] [Q&A]
Imagine i want to edit one Item and validate if the new name already exists on server. I need to know the Id of the current item, so i can ignore it when comparing.

Is there any way to use other properties from the current item in the validator? Or whats the best way to access the current EditItem?

**Comment by stsrki** [comment]
Yes, that's possible. Validator is just a function that you can write by yourself. If you want to check for your current Id, the best solution is to get it from the selected row item with binding:  `@bind-SelectedRow="@selectedEmployee"`. 

And then have a validator, like this:

```cs
public void CheckEMail( ValidatorEventArgs validationArgs )
{
    // check selectedEmployee.Id here

    ValidationRule.IsEmail( validationArgs );

    if ( validationArgs.Status == ValidationStatus.Error )
    {
        validationArgs.ErrorText = "EMail has to be valid email";
    }
}
```

**Reply by paulosdfigueiredo** [reply]
That's the workaround i am using, but it's not totally correct. 

Edit is different from Select, i can be editing an item that's not selected.

![image](https://user-images.githubusercontent.com/50837250/102996097-5d32ab00-451a-11eb-8ba9-463d5be62b5b.png)
Using the Demo, i am editing Id 5 while 4 is selected.
